apps/authorization/management/commands/repair_grants_expiration_date.py

```diff
@@ -0,0 +1,182 @@
+            print("Error: appname does not match appnameagain.")
+            return False
+
+        # validate created before date
+        turn_on_date_str = None
+
+        if options['turnondate']:
+            turn_on_date_str = options['turnondate']
+
+        if turn_on_date_str is None:
+            print("Error: turnondate required to repair grants.")
+            return False
+
+        turn_on_date = None
+        
+        try:
+            turn_on_date = datetime.strptime(turn_on_date_str, DATETIME_FMT).replace(tzinfo=timezone.utc)
+        except ValueError as e:
+            print("Error: bad date time value: {}".format(turn_on_date_str))
+            print(e)
+            return False
+
+        # validate created before date
+        created_before_date_str = None
+        
+        if options['createdbefore']:
+            created_before_date_str = options['createdbefore']
+
+        if created_before_date_str is None:
+            print("Error: createdbefore required to repair grants.")
+            return False
+
+        created_before_date = None
+        
+        try:
+            created_before_date = datetime.strptime(created_before_date_str, DATETIME_FMT).replace(tzinfo=timezone.utc)
+        except ValueError as e:
+            print("Error: bad date time value: {}".format(created_before_date_str))
+            print(e)
+            return False
+
+        current_dt = datetime.now().replace(tzinfo=pytz.UTC)
+
+        # time range must be a past time
+        if created_before_date > current_dt:
+            print("Error: createdbefore date must be a past time, createdbefore = {}, current date time ={}".format(created_before_date, current_dt))
+            return False
+
+        # validate created after date
+        created_after_date_str = None
+
+        if options['createdafter']:
+            created_after_date_str = options['createdafter']
+
+        if created_after_date_str is None:
+            print("Error: createdafter required to repair grants.")
+            return False
+
+        created_after_date = None
+        
+        try:
+            created_after_date = datetime.strptime(created_after_date_str, DATETIME_FMT).replace(tzinfo=timezone.utc)
+        except ValueError as e:
+            print("Error: bad date time value: {}".format(created_after_date_str))
+            print(e)
+            return False
+
+        # validate date range
+        if created_after_date > created_before_date:
+            print("Error: createdbefore must >= createdafter : createdbefore = {}, createdafter = {}".format(created_before_date_str, created_after_date_str))
+            return False
+
+        # validate turn on date in date range
+        if not (created_after_date <= turn_on_date and turn_on_date <= created_before_date):
+            print("Error: turnondate {} must be in [createdafter {}: createdbefore{}]".format(turn_on_date_str, created_after_date_str, created_before_date_str))
+            return False
```

Comment:
I don't understand the reason for this restriction, the created before/after seems to just be used to determine which grants to update, and the turnondate seems to only be used to determine what the new expiration date should be, so it would still be valid to use those parameters, even if the turnondate is outside of that range.